### PR TITLE
Add some mappings for GUIs and input handling

### PR DESCRIPTION
**Pull request by SquidDev**
- Add keyboard, mouse and clipboard classes. The names and locations are terrible. It may be worth having a `client.input` package or something.
 - Flesh out the GuiEventListener mappings.

I'm happy to remove some things - this was just the end result of a mod-porting session, so not everything may be entirely sane.

### CODE REVIEW

File: mappings/net/minecraft/client/gui/widget/WidgetListVideoSettings.mapping [hunk 1]
```diff
@@ -6,11 +6,15 @@ CLASS cox net/minecraft/client/gui/widget/WidgetListVideoSettings
 		METHOD a drawEntry (IIIIZF)V
 		METHOD mouseClicked (DDI)Z
 			ARG 0 mouseX
-			ARG 1 mouseY
```
Comment (SquidDev) [inline mappings/net/minecraft/client/gui/widget/WidgetListVideoSettings.mapping:9]: I'm really not sure what's going on here - it's like one tool was started at 0, and another at 1. Similarly why there's a `keyCode` argument here at all.

I'm happy to clean this up, just would like someone else to look over it first.

File: mappings/net/minecraft/util/ChatCharacterFilter.mapping [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CLASS k
+CLASS k net/minecraft/util/ChatCharacterFilter
```
Comment (jamierocks) [inline mappings/net/minecraft/util/ChatCharacterFilter.mapping]: Judging from realm's `RealmsSharedConstants`, this is more likely `net/minecraft/SharedConstants`. It'll be the obfuscator removing fields that are inlined, as to why it doesn't fell much like a typically constants class.

**Review by jamierocks — COMMENTED** [review]
Looks good :+1:

**Review by jamierocks — APPROVED** [review]

**Comment by jamierocks** [comment]
Just need to look into the duplicate `mouseX` arguments.

**Comment by asiekierka** [comment]
That's likely to be Enigma bugfix artifacts, don't worry about it.

**Pull request merged**